tooling/src/main/java/org/opencds/cqf/tooling/casereporting/tes/TESPackageGenerator.java

```diff
@@ -0,0 +1,615 @@
+        int rowCounter = 0;
+        Row headerRow = sheet.createRow(rowCounter);
+        Cell headerSystemCell = headerRow.createCell(0);
+        headerSystemCell.setCellValue("System");
+        Cell headerCodeCell = headerRow.createCell(1);
+        headerCodeCell.setCellValue("Code");
+        Cell headerDisplayCell = headerRow.createCell(2);
+        headerDisplayCell.setCellValue("Text");
+        rowCounter++;
+
+        for (int i = 0; i < baseList.size(); i++) {
+            String code = baseList.get(i).getCode();
+            if (comparisonList.stream().noneMatch(rscoding -> rscoding.getCode().equalsIgnoreCase(code))) {
+                Row row = sheet.createRow(rowCounter);
+                Cell systemCell = row.createCell(0);
+                systemCell.setCellValue(baseList.get(i).getSystem());
+                Cell codeCell = row.createCell(1);
+                codeCell.setCellValue(baseList.get(i).getCode());
+                Cell displayCell = row.createCell(2);
+                displayCell.setCellValue(baseList.get(i).getDisplay());
+                rowCounter++;
+            }
+        }
+    }
+
+    private static List<Coding> extractFlatCodeListFromValueSet(ValueSet conditionCodeValueSet) {
+        List<Coding> conditionCodeValueSetCodeAndDisplay = new ArrayList<>();
+
+        if (conditionCodeValueSet.hasCompose() && conditionCodeValueSet.getCompose().hasInclude()) {
+            for (ValueSet.ConceptSetComponent include : conditionCodeValueSet.getCompose().getInclude()) {
+                for (ValueSet.ConceptReferenceComponent concept : include.getConcept()) {
+                    conditionCodeValueSetCodeAndDisplay.add(new Coding(include.getSystem(), concept.getCode(), concept.getDisplay()));
+                }
+            }
+        }
+
+        return conditionCodeValueSetCodeAndDisplay;
+    }
+
+    private ValueSet loadConditionCodeValueSet(String pathToConditionCodeValueSet) {
+        ValueSet conditionCodeValueSet = null;
+        File valueSetFile = new File(pathToConditionCodeValueSet);
+        if (valueSetFile.isFile()) {
+            try {
+                if (valueSetFile.getName().endsWith("json")) {
+                    conditionCodeValueSet = (ValueSet)((JsonParser) fhirContext.newJsonParser()).parseResource(new FileInputStream(valueSetFile));
+                }
+                else if (valueSetFile.getName().endsWith("xml")) {
+                    conditionCodeValueSet = (ValueSet)((XmlParser) fhirContext.newXmlParser()).parseResource(new FileInputStream(valueSetFile));
+                }
+                else {
+                    throw new IllegalArgumentException("Unsupported input file encoding. Currently, only .json and .xml supported for the input file.");
+                }
+            } catch (FileNotFoundException e) {
+                e.printStackTrace();
+                throw new RuntimeException("Error parsing " + valueSetFile.getName());
```

Comment:
Not really doing anything with this catch. Typically you want to log a warning or throw a meaningful error, but not both. Also generic exceptions should be used sparingly